### PR TITLE
ST: Check rack-warnes configuration for node affinity

### PR DESCRIPTION
**Pull request by Frawless**
Signed-off-by: Jakub Stejskal <xstejs24@gmail.com>

### Type of change

- Enhancement / new feature

### Description

Test enhancement which should cover this https://github.com/strimzi/strimzi-kafka-operator/pull/2347.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Frawless** [comment]
@strimzi-ci run tests testcase=SpecificST#testRackAware profile=specific test_only

File: systemtest/src/test/java/io/strimzi/systemtest/specific/SpecificST.java [hunk 1]
```diff
@@ -59,15 +56,25 @@ void testRackAware() throws Exception {
...
+        assertThat(kafkaPodAffinityTerm.getTopologyKey(), is(rackKey));
+        assertThat(kafkaPodAffinityTerm.getLabelSelector().getMatchLabels(), hasEntry("strimzi.io/cluster", CLUSTER_NAME));
+        assertThat(kafkaPodAffinityTerm.getLabelSelector().getMatchLabels(), hasEntry("strimzi.io/name", KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)));
+
         String rackId = cmdKubeClient().execInPod(KafkaResources.kafkaPodName(CLUSTER_NAME, 0), "/bin/bash", "-c", "cat /opt/kafka/init/rack.id").out();
         assertThat(rackId.trim(), is("zone"));
 
-        String brokerRack = cmdKubeClient().execInPod(KafkaResources.kafkaPodName(CLUSTER_NAME, 0), "/bin/bash", "-c", "cat /tmp/strimzi.properties | grep broker.rack").out();
-        assertThat(brokerRack.contains("broker.rack=zone"), is(true));
+        String brokerRack = cmdKubeClient().execInPod(KafkaResources.kafkaPodName(CLUSTER_NAME, 0), "/bin/bash", "-c", "cat /tmp/strimzi.properties | grep rack.id").out();
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/specific/SpecificST.java]: I think you found a bug here ... so you should not change the test accept the bug ;-). I opened #2550 to fix it.

Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/specific/SpecificST.java]: hehe, I will change it back then :)

**Comment by ppatierno** [comment]
@Frawless fixes the conflicts please

**Comment by Frawless** [comment]
Cherry-pick to `release-0.17.x`

**Comment by scholzj** [comment]
@Frawless Can you do one moree rebase of this? It looks like the previous PR created some conflict. This should be hopefully next in the pipeline.

**Comment by Frawless** [comment]
Conflicts resolved, Cherry-pick to `release-0.17.x` please

**Pull request merged**